llvm/lib/Target/AArch64/GISel/AArch64RegisterBankInfo.cpp

```diff
@@ -774,8 +793,12 @@ AArch64RegisterBankInfo::getInstrMapping(const MachineInstr &MI) const {
              (getRegBank(ScalarReg, MRI, TRI) == &AArch64::FPRRegBank ||
               onlyDefinesFP(*ScalarDef, MRI, TRI)))
       OpRegBankIdx = {PMI_FirstFPR, PMI_FirstFPR};
-    else
+    else {
+      if (ScalarTy.getSizeInBits() < 32 &&
+          getRegBank(ScalarReg, MRI, TRI) == &AArch64::GPRRegBank)
+        MappingID = 1;
```

Comment:
Can you make `1` here a `const unsigned` with a name like `CustomID` so it's clearer? Also add a comment here so the reader knows it causes `applyMappingImpl` to be called.